src/Evolution/Actions/BackgroundGrVars.hpp

```diff
@@ -0,0 +1,184 @@
+ *
+ */
+template <typename System, bool used_in_evolution_phase>
+struct BackgroundGrVars {
+  using simple_tags_from_options = tmpl::list<::Tags::Time>;
+
+  // Collect all the GR quantities used in the templated evolution system
+  using gr_variables_tag = ::Tags::Variables<tmpl::remove_duplicates<
+      tmpl::append<typename System::spacetime_variables_tag::tags_list,
+                   typename System::flux_spacetime_variables_tag::tags_list,
+                   tmpl::list<typename System::inverse_spatial_metric_tag>>>>;
+
+  using simple_tags = tmpl::list<gr_variables_tag>;
+  using compute_tags = tmpl::list<>;
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static Parallel::iterable_action_return_t apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ArrayIndex& /*array_index*/, ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    // Note : once every evolution system is refactored to use the runtime
+    // initial data, we can just use
+    // `evolution::initial_data::Tags::Initialdata` to retrieve the analytic
+    // solution or data. Then we may be able to convert this action into a
+    // mutator or an argument of Initialization::Actions::AddSimpleTags<>.
+    //
+    if constexpr (db::tag_is_retrievable_v<
+                      evolution::initial_data::Tags::InitialData,
+                      db::DataBox<DbTagsList>>) {
+      using derived_classes =
+          tmpl::at<typename Metavariables::factory_creation::factory_classes,
+                   evolution::initial_data::InitialData>;
+      call_with_dynamic_type<void, derived_classes>(
+          &db::get<evolution::initial_data::Tags::InitialData>(box),
+          [&box](const auto* const data_or_solution) {
+            impl(make_not_null(&box), *data_or_solution);
+          });
+    } else if constexpr (db::tag_is_retrievable_v<
+                             ::Tags::AnalyticSolutionOrData,
+                             db::DataBox<DbTagsList>>) {
+      impl(make_not_null(&box), db::get<::Tags::AnalyticSolutionOrData>(box));
+    } else {
+      ERROR(
+          "Either ::Tags::AnalyticSolutionOrData or "
+          "evolution::initial_data::Tags::InitialData must be in the "
+          "DataBox.");
+    }
+    return {Parallel::AlgorithmExecution::Continue, std::nullopt};
+  }
+
+ private:
+  template <typename DbTagsList, typename T>
+  static void impl(const gsl::not_null<db::DataBox<DbTagsList>*> box,
+                   const T& solution_or_data) {
+    static constexpr size_t Dim = System::volume_dim;
+
+    using GrVars = typename gr_variables_tag::type;
+
+    if constexpr (used_in_evolution_phase) {
+      // Check if the mesh is actually moving i.e. block coordinate map is
```

Comment:
what about the case where the solution is time-dependent, but the mesh is not moving?

Reply:
hmm, I don't see a way quickly browsing.  For now just add a warning to the documentation of the action that it assumes the solution is time-independent, and create an issue to remind us to fix this at some point.  Probably doesn't make sense to do that until we can create initial data from the input file for all systems.

Reply:
Okay. I also think this would not be a serious practical problem unless we want to use time-dependent GR solutions as a background, which I can't think of many.